inlong-sort/sort-connectors/iceberg/src/main/java/org/apache/inlong/sort/iceberg/sink/multiple/IcebergMultipleStreamWriter.java

```diff
@@ -246,55 +246,64 @@ public void processElement(RecordWithSchema recordWithSchema) throws Exception {
+                    long size = data == null ? 0 : data.toString().getBytes(StandardCharsets.UTF_8).length;
 
-                try {
-                    multipleWriters.get(tableId).processElement(data);
-                } catch (Exception e) {
-                    LOG.error(String.format("write error, raw data: %s", data), e);
-                    if (!dirtyOptions.ignoreDirty()) {
-                        throw e;
-                    }
-                    if (dirtySink != null) {
-                        DirtyData.Builder<Object> builder = DirtyData.builder();
-                        try {
-                            String dirtyLabel = DirtySinkHelper.regexReplace(dirtyOptions.getLabels(),
-                                    DirtyType.BATCH_LOAD_ERROR, null,
-                                    dataBaseName, tableName, null);
-                            String dirtyLogTag =
-                                    DirtySinkHelper.regexReplace(dirtyOptions.getLogTag(),
-                                            DirtyType.BATCH_LOAD_ERROR, null,
-                                            dataBaseName, tableName, null);
-                            String dirtyIdentifier =
-                                    DirtySinkHelper.regexReplace(dirtyOptions.getIdentifier(),
-                                            DirtyType.BATCH_LOAD_ERROR, null,
-                                            dataBaseName, tableName, null);
-                            builder.setData(data)
-                                    .setLabels(dirtyLabel)
-                                    .setLogTag(dirtyLogTag)
-                                    .setIdentifier(dirtyIdentifier)
-                                    .setRowType(multipleWriters.get(tableId).getFlinkRowType())
-                                    .setDirtyMessage(e.getMessage());
-                            dirtySink.invoke(builder.build());
-                            if (sinkMetricData != null) {
-                                sinkMetricData.outputDirtyMetricsWithEstimate(dataBaseName,
-                                        tableName, 1, size);
-                            }
-                        } catch (Exception ex) {
-                            if (!dirtyOptions.ignoreSideOutputErrors()) {
-                                throw new RuntimeException(ex);
+                    try {
+                        multipleWriters.get(tableId).processElement(data);
+                    } catch (Exception e) {
+                        LOG.error(String.format("write error, raw data: %s", data), e);
+                        if (!dirtyOptions.ignoreDirty()) {
+                            throw e;
+                        }
+                        if (dirtySink != null) {
+                            DirtyData.Builder<Object> builder = DirtyData.builder();
+                            try {
+                                String dirtyLabel = DirtySinkHelper.regexReplace(dirtyOptions.getLabels(),
```

Comment:
would be better if the construction for dirty data can be extraced since the method processElement is too long for reading

Reply:
Fixed